parsl/executors/taskvine/executor.py

```diff
@@ -232,145 +114,173 @@ class TaskVineExecutor(BlockProviderExecutor, putils.RepresentationMixin):
+        # tasks' futures to done status.
+        self.collector_thread = None
+
+        # track task id of submitted parsl tasks
+        # task ids are incremental and start from 0
+        self.executor_task_counter = 0
+
+        # track number of tasks that are waiting/running
+        self.outstanding_tasks = 0
+
+        # Lock for threads to access self.outstanding_tasks attribute
+        self.outstanding_tasks_lock = threading.Lock()
 
-        if not self.address:
-            self.address = socket.gethostname()
+        # Threading lock to manage self.tasks dictionary object, which maps a task id
+        # to its future object.
+        self.tasks_lock = threading.Lock()
+
+        # Worker command to be given to an execution provider (e.g., local or Condor)
+        self.worker_command = ""
 
-        if self.project_password_file is not None and not os.path.exists(self.project_password_file):
-            raise TaskVineFailure('Could not find password file: {}'.format(self.project_password_file))
+        # Path to directory that holds all tasks' data and results, only used when
+        # manager's task mode is 'regular'.
+        self.function_data_dir = ""
 
-        # Build foundations of the launch command
-        self.launch_cmd = ("python3 exec_parsl_function.py {mapping} {function} {result}")
-        if self.init_command != "":
-            self.launch_cmd = self.init_command + "; " + self.launch_cmd
+        # helper scripts to prepare package tarballs for Parsl apps
+        self.package_analyze_script = shutil.which("poncho_package_analyze")
+        self.package_create_script = shutil.which("poncho_package_create")
 
     def _get_launch_command(self, block_id):
-        # this executor uses different terminology for worker/launch
-        # commands than in htex
+        # Implements BlockProviderExecutor's abstract method.
+        # This executor uses different terminology for worker/launch
+        # commands than in htex.
         return f"PARSL_WORKER_BLOCK_ID={block_id} {self.worker_command}"
 
+    def __synchronize_manager_factory_comm_settings(self):
+        # Synchronize the communication settings between the manager and the factory
+        # so the factory can direct workers to contact the manager.
+
+        # If the manager can choose any available port (port number = 0),
+        # then it must have a project name
+        # so the factory can look it up. Otherwise the factory process will not know the
+        # port number as it's only chosen when the TaskVine manager process is run.
+        if self.manager_config.port == 0 and self.manager_config.project_name is None:
+            self.manager_config.project_name = "parsl-vine-" + str(uuid.uuid4())
+
+        # guess the host name if the project name is not given
+        if not self.manager_config.project_name:
+            self.manager_config.address = socket.gethostname()
```

Comment:
there is a module `parsl.addresses` which provides various methods for working out hostnames, intended for use with the `address` parameter of the high throughput executor. This is because we haven't found one method that works everywhere. That experience makes me think that eventually that might be needed here too (although that doesn't need to block this PR being merged - it's a note for the future)